securedrop_client/logic.py

```diff
@@ -479,10 +494,10 @@ def logout(self):
         Call logout function in the API, reset the API object, and force the UI
         to update into a logged out state.
         """
-        self.call_api(self.api.logout,
-                      self.on_logout_success,
-                      self.on_logout_failure)
-        self.api = None
+        if self.api is not None:
+            self.call_api(self.api.logout, self.on_logout_success, self.on_logout_failure)
```

Comment:
gotcha, so if `self.api` is already None, we don't need to invalidate the token, because the token is already invalid/expired? (from inspecting the logic in `on_sync_failure` and `on_queue_paused`). 

i think it would be good to write that this is what's going on in a comment somewhere (as it's only obvious from tracing the logic and could be easily missed in future modifications). alternatively we could move the setting of `self.api = None` into this method and make it more explicit that this is what's happening via an `invalidate_token` kwarg or something. Let me know if you have a better idea here.

Reply:
i think only setting api to none in one place is a good idea. i was thinking about how I'd like to refactor how we check whether or not we're in offline_mode vs just _offline_ (no token) but i think make a small change to only set api to None in one place would be a good addition to this pr.

Reply:
more on the whole offline-mode vs no-token: this is relevant if we want to provide a way for users to regenerate a new token without losing jobs in the api queue (so just pause and wait for a new token), but this is definitely a post-pilot issue